src/main/java/com/sforce/cd/apexUnit/client/utils/ApexClassFetcherUtils.java

```diff
@@ -321,8 +322,8 @@ public static String fetchAndAddToMapApexClassIdBasedOnName(PartnerConnection co
 		}
 		if (queryResult != null && queryResult.getDone()) {
 			String[] classIds = fetchApexClassesAsArray(queryResult);
-			for (String classId : classIds) {
-				apexClassId = classId;
+			if(classIds.length>0){
+				apexClassId = classIds[0];
```

Comment:
@adarsh-ramakrishna-sfdc  I took the liberty of assuming that there are no **two** classes with same name in a single namespace. And then found that if we try creating another class with the same name in a namespace, there is an error.